preferences/tests.py

```diff
@@ -342,3 +342,138 @@ def test_preferences_record_update_POST__bad_request(self):
+            'last_name': 'Doe',
+            'email': 'johnd123@uvic.ca',
+            'is_superuser': False
+        }
+        self.user = User.objects.create_user(**self.user_attributes)
+
+        self.app_user_attributes = {
+            'user': self.user,
+            'prof_type': 'RP',
+            'is_peng': True
+        }
+        self.app_user = AppUser.objects.create(**self.app_user_attributes)
+        self.app_user_serializer = AppUserSerializer(instance=self.app_user)
+
+        #create associated Preferences record
+        self.preferences_attributes = {
+            "professor": self.app_user,
+            "is_submitted": True,
+            "is_unavailable_sem1": False,
+            "is_unavailable_sem2": True,
+            "num_relief_courses": 1,
+            "taking_sabbatical": True,
+            "sabbatical_length": "FULL",
+            "sabbatical_start_month": 1,
+            "preferred_hours": [
+                {"Mon": "8am-9am"},
+                {"Thu": "1pm-2pm"}
+            ],
+            "teaching_willingness": {
+                "CSC226": "Very Willing"
+            },
+            "teaching_difficulty": {
+                "CSC226": "Able"
+            },
+            "wants_topics_course": True,
+            "topics_course_id": "CSC485c",
+            "topics_course_name": "Data Management and Parallelization"
+        }
+
+        #provide some default Preferences data to be used as a request body
+        self.default_serializer_data = {
+            'professor': 'johnd1',
+            'is_submitted': True,
+            'is_unavailable_sem1': False,
+            'is_unavailable_sem2': True,
+            'num_relief_courses': 1,
+            'taking_sabbatical': True,
+            'sabbatical_length': 'FULL',
+            'sabbatical_start_month': 1,
+            'preferred_hours': [
+                {'Mon': '8am-9am'},
+                {'Thu': '1pm-2pm'}
+            ],
+            'teaching_willingness': {
+                'CSC226': 'Very Willing'
+            },
+            'teaching_difficulty': {
+                'CSC226': 'Able'
+            },
+            'wants_topics_course': True,
+            'topics_course_id': 'CSC485c',
+            'topics_course_name': 'Data Management and Parallelization'
+        }
+
+
+    @classmethod
+    def save_default_user(self):
+        serializer = AppUserSerializer(data=self.default_serializer_data)
+        if serializer.is_valid():
+            serializer.save()
+
+    @classmethod
+    def get_nonadmin_API_client(self): 
+        user = self.user
+        client = APIClient()
+        token = SlidingToken.for_user(user)
+        client.credentials(HTTP_AUTHORIZATION=f'Bearer {token}')
+        return client
+
+    @classmethod
+    def post_default_user(self) -> HttpResponse: 
+        client = self.get_nonadmin_API_client()
+        response: HTTPResponse = client.post("/api/preferences/", data=self.default_serializer_data, format='json')
```

Comment:
Should all these response types be `HttpResponse` instead of `HTTP..`? Tests seem to pass but my VSCode yells at the typing being unknown

Reply:
Ah yes. I've been changing a lot of things in my dev environment and haven't gotten my linter working quite right yet. I'll change that.